### DISCUSSION TITLE
Is there any replacement for New XKit's "Audio Downloader"?

### DISCUSSION THREAD

**Discussion by arcanetrivia** [AprilSylph/XKit-Rewritten#1669] [Lounge]
I've been gradually disabling/uninstalling extensions from New XKit for a while, with an eye to completely uninstalling the whole thing from Firefox (especially now that Mozilla has started automatically disabling the extension in the browser because it doesn't meet their guidelines of having the source available, or something). The last one that seems to still do anything in the first place (many of its options have no effect now) that I can't replace with equivalent functionality either native or via XKit Rewritten, is Audio Downloader. This puts a link under the audio player to download the associated audio file: 
![image](https://github.com/user-attachments/assets/6b505f27-26fe-4736-9cd7-e2e5d358d00c)
It's possible to get there by inspecting the page element and finding the audio file URL in the source, but having it right there for one-click is convenient. I don't see anything like this in XKit Rewritten's extensions.

It's ok if this is a "wontfix" as it's such a minor thing, but I just wondered if I missed it somewhere in Rewritten.

**Comment by marcustyphoon** [comment]
I believe XKit Rewritten's Vanilla Audio includes a download button in the three-dots menu—it might depend on the browser, of course, since the whole point is to use the built-in browser audio element. Does that work on yours?

**Reply by AprilSylph** [reply]
On Firefox, you instead have to right-click the vanilla audio element to get the menu with the download option.

![image](https://github.com/user-attachments/assets/b0c41150-ef25-498d-ad4f-fe6566fa1ae7)

**Comment by arcanetrivia** [comment]
Ok, I guess that can work. I do have to turn on Vanilla Audio to see that, otherwise the context menu is different. But that will probably serve my purpose. Welp, I guess it's goodbye New XKit at long last, then! Thanks for the replies.
